arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -292,7 +258,11 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
         std::shared_ptr<LogicalCollection> ci;
         try {
           ci = ClusterInfo::instance()->getCollection(
-            std::dynamic_pointer_cast<LogicalCollection>(it->second)->dbName(),
+            #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+              std::dynamic_pointer_cast<LogicalCollection>(it->second)->dbName(),
```

Comment:
better
```
#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
   dynamic_cast<LogicalCollection&>(*it->second).dbName()
   ...
#endif
```
since it's going to throw meaningful exception in case of error